UnityProject/Assets/Scripts/Map/Asteroid.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Networking;
+
+public class Asteroid : NetworkBehaviour {
+
+	private MatrixMove mm;
+
+	private float asteroidDistance = 550; //How far can asteroids be spawned
+
+	private float distanceFromStation = 175; //Offset from station so it doesnt spawn into station
+
+
+	private void Start()
+	{
+		mm = GetComponent<MatrixMove>();
+
+		SpawnNearStation();
+	}
+
+	[Server]
+	public void SpawnNearStation()
+	{
+		if (isServer)
+		{
+			//Based on EscapeShuttle.cs
+			mm.SetPosition(Random.insideUnitCircle * asteroidDistance + new Vector2(distanceFromStation, -distanceFromStation));
```

Comment:
This won't work:
That `Vector2(distanceFromStation, -distanceFromStation)` is actually **always** Vector2( x=175; y=-175 ),
which equals to bottom right
So it doesn't guarantee that it won't spawn inside station, as first part can generate -175; 175 and asteroid will land exactly at the station, so please test your calculations before PRing!
What you need to do is exclude the entire range of ( x=-175..175; y=-175..175 ) from final randomised position by per-axis `Mathf.Clamp`ing or something else

Reply:
Also it would be nice if asteroids had random rotation on start

Reply:
...And some could be moving very slowly